### DISCUSSION TITLE
How to merge two GriefPrevention databases?

### DISCUSSION THREAD

**Discussion by AxlRose-RX** [GriefPrevention/GriefPrevention#2129] [Q&A]
- I have 3 old survival servers under the same network, and I want to end up with only one
- I want to merge the 3 of them into one to clear resources and stop maintaining old servers that not many people play anymore.
- The 3 servers have it's own MySQL data with data from GriefPrevention.
- I will import with Multiverse the old worlds from the 3 old survivals and I want to move the claims too into a new "Legacies" server database for the new "Legacies" server
- How can I merge the 3 databases into one? is there any tool for this? can maybe go from MySQL to File and then from File import to new database? 

- I was going to try to manually move the tables but no idea how, as some  "id" numbers are the same from the different servers and there is also on all griefprevention_nextclaimid and also griefprevention_playerdata tables. Some of my players played on the first survival and they still maintain their claims from the old survivals as they also do on the newer survival servers on my network.

**Comment by QarthO** [comment]
You'd need to do this manually, AFAIK there is no native way in GP to do such a thing.

A complex option: You can merge after writing a script to manually go thru the old tables and add a value to the id of each of the claims to sync up with your main server. (ie. if the most recent id on the main server is 234, then add 234 to all the ids on the old server. Then make sure u change the next id on the main server to match up to how many total claims there are, so new claims aren't conflicting and have the same id

IMO, the easiest option would be to not transfer over the claims. Instead, just tell ur players who are active to go and re-claim their stuff. It's not a hassle for them to reclaim them.

**Comment by AxlRose-RX** [comment]
I think I figure it out, export the database to text backup .sql then run script over to modify the ids with regex from each beginning of line after ( and before , something like (3139,

Then find the player name UUID on from playerdata and keep the latest lastlogin and add the accruedblocks and bonusblocks

Then with the new text .sql file import it
